app/facades/webhooks.py

```diff
@@ -0,0 +1,71 @@
+import json
+import base64
+from typing import Union
+import os
+from typing import List, Literal
+from aries_cloudcontroller import AcaPyClient
+
+from httpx import AsyncClient, get, HTTPError
+
+
+WEBHOOKS_URL = os.getenv("WEBHOOKS_URL", "http://yoma-webhooks-web:3010")
+ADMIN_API_KEY = os.getenv("ACAPY_ADMIN_API_KEY", "adminApiKey")
+
+topics = Literal[
+    "connections",
+    "issue_credential",
+    "forward",
+    "ping",
+    "basicmessages",
+    "issuer_cred_rev",
+    "issue_credential_v2_0",
+    "issue_credential_v2_0_indy",
+    "issue_credential_v2_0_dif",
+    "present_proof",
+    "revocation_registry",
+]
+
+
+def get_wallet_id_from_client(client: Union[AcaPyClient, AsyncClient]) -> str:
```

Comment:
A problem with this approach is that we can't verify if the JWT is valid. For all other requests we're calling ACA-Py which will verify whether the JWT is valid. 

I can just create a JWT with a wallet id and you can retrieve all webhooks for a specific wallet: https://jwt.io/#debugger-io?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ3YWxsZXRfaWQiOiJteV93YWxsZXRfaWQifQ.9ALYlsIGqjfzP1s2EzUa0tCIjWWPElAyKmlsrPIzeOw

You still need the base adminApiKey, so not that much of a problem in this case, but not sure if we're creating a change for security issues here

Reply:
I see what you mean, but there is two reasons why I thought that not a problem:

1. Thsi method is not exposed to the user directly and relies on extracting the token from controller/client which gets instantiated via fastapi mechanism
2. The fastapi can remain the gatekeeper (as we do know already).

If wanted I can also add a verification of the jwt?!